pkg/secrets/manager.go

```diff
@@ -48,13 +48,7 @@ func AreCompatible(a, b Manager) bool {
 		return false
 	}
 
-	as, err := json.Marshal(a.State())
-	if err != nil {
-		return false
-	}
-	bs, err := json.Marshal(b.State())
-	if err != nil {
-		return false
-	}
+	as := a.State()
+	bs := b.State()
 	return string(as) == string(bs)
```

Comment:
We don't have to convert to string here. We can use `bytes.Equal`.

```suggestion
	return bytes.Equal(as, bs)
```